src/QuadGK.jl

```diff
@@ -160,6 +143,31 @@ function do_quadgk(f, s, n, ::Type{Tw}, abstol, reltol, maxevals, nrm) where Tw
     return (I, E)
 end
 
+function preprocess_real(f, s::TS, n, ::Type{Tw}, abstol, reltol, maxevals, nrm) where {Tw,ET<:Real,TS<:AbstractVector{ET}}
+    s1 = s[1]; s2 = s[end]; inf1 = isinf(s1); inf2 = isinf(s2)
+    if inf1 || inf2
+        if inf1 && inf2 # x = t/(1-t^2) coordinate transformation
+            return do_quadgk_common(t -> begin t2 = t*t; den = 1 / (1 - t2);
+                                        f(t*den) * (1+t2)*den*den; end,
+                                [isinf(x) ? copysign(one(x), x) : 2x / (1+hypot(1,2x)) for x in s],
+                                n, Tw, abstol, reltol, maxevals, nrm)
+        end
+        (s0,si) = inf1 ? (s2,s1) : (s1,s2)
+        if si < 0 # x = s0 - t/(1-t)
+            return do_quadgk_common(t -> begin den = 1 / (1 - t);
```

Comment:
You can use a `do` block rather than `t -> begin`.

Reply:
it was written like this, I found it a bit dense, I'll rewrite this part